iothub/device/src/Transport/Amqp/ClientWebSocketTransport.cs

```diff
@@ -59,8 +59,8 @@ public override bool WriteAsync(TransportAsyncCallbackArgs args)
         {
             ThrowIfNotOpen();
 
```

Comment:
Given we're in an internal file for transport, when would these conditions fail and what would an exception mean to the user if they observed it? What would they do differently in their code to avoid it?